lang/macro/src/codegen/events.rs

```diff
@@ -119,7 +119,7 @@ impl EventHelpers<'_> {
                     where
                         E: Into<Self::Event>,
                     {
-                        ink_lang::EnvAccess::<EnvTypes>::emit_event(self, event.into())
+                        ink_lang::EnvAccess::<EnvTypes>::emit_event_inner(self, event.into())
```

Comment:
I think it could already help if we instead do:
```rust
ink_lang::EnvAccess::<EnvTypes>::emit_event::<Self::Event>(self, event.into())
```

Reply:
The problem is that calling `self.env().emit_event()` from the contract was calling the wrong `emit_event`. It was calling the inherent one rather than the `impl EmitEvent`